### ISSUE TITLE
Negative loss? is that a problem?

### ISSUE THREAD

**Issue by avinash31d** [andreas128/SRFlow#12]
I followed this repo, paper and glow paper. My loss looks like this

        **_logp =  GaussianDiag.logp(None, None, (z))
        obj = logdets + logp
        loss = -obj / (tf.math.log(2.) * tf.cast(pixels, tf.float32))_**

There are 2 terms added together : one is **logp** which is always negative, and **logdets** which starts with a negative number during training, because of the minus sign in **-obj** the loss is positive, but after training for around 100k steps, both the **logp** and **logdets** increases. The logp term increases and becomes positive number and hence the final loss becomes negative around **-3.xxx**. Just wanted to know whether it is an expected behaviour ?

**Comment by andreas128** [comment]
The final loss of SRFlow is negative and steadily decreasing:

<img width="640" alt="PyTorch Normalizing Flow Loss" src="https://user-images.githubusercontent.com/11280511/105580348-22779900-5d8c-11eb-8703-b2fe714c277c.png">

We did not track those two components separately. Please try to run our code and log the loss (NLL is defined [here](https://github.com/andreas128/SRFlow/blob/205e63f1f96f8d98956c4a1034c6cfa04c6db8bc/code/models/modules/SRFlowNet_arch.py#L112)).

Did that help?

**Comment by avinash31d** [comment]
Hi @andreas128,

Thanks, that is something I was looking for and it clears my doubt about the logdets term.
